src/Optimizer/AbstractOptimizer.php

```diff
@@ -90,4 +92,15 @@ protected function typeIsString($data): bool
     {
         return isset($data['type']) && true === is_string($data['type']);
     }
+
+    /**
+     * @param array|mixed $data
+     * @return bool
+     */
+    protected function isPrimitive($data): bool
```

Comment:
I am thinking, it would be nice to just have one function, so either pass an optional arg (doesn't need to be in the interface), to optimize or we move the `isPrimitive` function to the `Avro` class

Reply:
as mentioned above, you can probably kill this func now and just pass `$resolvedTemplate->isPrimitive()` to `optimize` in the merger :v: